packages/server/lib/webhook/gmail-webhook-routing.ts

```diff
@@ -0,0 +1,128 @@
+import crypto from 'node:crypto';
+
+import axios from 'axios';
+
+import { NangoError, environmentService, getGlobalWebhookReceiveUrl } from '@nangohq/shared';
+import { Err, Ok, getLogger } from '@nangohq/utils';
+
+import type { WebhookHandler } from './types.js';
+import type { LogContextGetter } from '@nangohq/logs';
+import type { Config as ProviderConfig } from '@nangohq/shared';
+
+const logger = getLogger('Webhook.Gmail');
+
+interface DecodedDataObject {
+    emailAddress: string;
+    historyId: string;
+}
+
+export async function validate(integration: ProviderConfig, headers: Record<string, any>): Promise<boolean> {
+    try {
+        const authHeader: string | undefined = headers['authorization'];
+
+        if (!authHeader) {
+            return true;
+        }
+
+        if (!authHeader?.startsWith('Bearer ')) {
+            return false;
+        }
+
+        const token = authHeader.split(' ')[1];
+        if (!token) {
+            logger.warning('No JWT token found in Authorization header');
+            return false;
+        }
+
+        const [headerB64, payloadB64, signatureB64] = token.split('.');
+        if (!headerB64 || !payloadB64 || !signatureB64) {
+            return false;
+        }
+
+        const header = JSON.parse(Buffer.from(headerB64, 'base64url').toString('utf8'));
+        const payload = JSON.parse(Buffer.from(payloadB64, 'base64url').toString('utf8'));
+        const signedData = `${headerB64}.${payloadB64}`;
+        const signature = Buffer.from(signatureB64, 'base64url');
+
+        const { data: jwks } = await axios.get('https://www.googleapis.com/oauth2/v3/certs');
```

Comment:
we will do one HTTP call per webhook?

Reply:
Updated with https://github.com/NangoHQ/nango/pull/4347/commits/1ecc2c66a5a96729a0034316102cc98be2c0ccfe